crates/viewer/re_space_view_dataframe/src/dataframe_ui.rs

```diff
@@ -284,21 +283,58 @@ impl<'a> egui_table::TableDelegate for DataframeTableDelegate<'a> {
                         }
                     };
 
+                    let header_ui = |ui: &mut egui::Ui| {
+                        let text = egui::RichText::new(column.short_name()).strong();
+
+                        let is_selected = match column {
+                            ColumnDescriptor::Time(descr) => {
+                                &descr.timeline == self.ctx.rec_cfg.time_ctrl.read().timeline()
+                            }
+                            ColumnDescriptor::Component(component_column_descriptor) => self
+                                .ctx
+                                .selection()
+                                .contains_item(&re_viewer_context::Item::ComponentPath(
+                                    component_column_descriptor.component_path(),
+                                )),
+                        };
+
+                        let response = ui.selectable_label(is_selected, text);
+
+                        match column {
+                            ColumnDescriptor::Time(descr) => {
+                                if response.clicked() {
+                                    self.ctx.command_sender.send_system(
+                                        re_viewer_context::SystemCommand::SetActiveTimeline {
+                                            rec_id: self.ctx.recording_id().clone(),
+                                            timeline: descr.timeline,
+                                        },
+                                    );
```

Comment:
I think the net change in behavior is an improvement, so I'm approving regardless, but this bit of functionality is somewhat unexpected for me as it's potentially disruptive to other views whereas setting the selection context in general is not.

Additionally this further highlights the peculiar fact that data table UI does not respect the currently selected timeline.